### PR TITLE
Setting up Github Actions and lightening Travis

### PR DESCRIPTION
**Pull request by lasofivec**
Three main changes to **tofu**'s CI/CD:

- Travis testing:
   On *all* branches two jobs: bionic 3.7 and trusty 3.6
   Only on `devel` and `master`: all jobs are done (bionic 3.7, trusty 3.6, xenial 3.7 and 3.6, osx 3.7 and 3.6)
- Github actions testing:
   Only on `devel` and `master`: [ubuntu-latest (rn: bionic), mac-latest (rn: osx 10.15), windows-latest (rn: )] x [3.6, 3.7, 3.8]
   On all branches: ubuntu latest with python 3.8
- Github actions deployment:
  Trying pypi seems to be working for test pypi. 
  To be able to test it on tofu for sure, we need to add two  add tokens for PyPi on repo's secret ([see this doc](https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#saving-credentials-on-github))

Quick fix for #452 
Solves #440

### CODE REVIEW

**Comment by flothesof** [comment]
Hi @lasofivec 
Just a quick question before I try it your PR on my side: do you plan on keeping travis builds on top of github actions? 
Isn't it better to just have GH actions run the CI on the three platforms and switch to just that?

**Comment by lasofivec** [comment]
I was thinking for the moment there is no problem.
Plus I haven't implemented all deployments with GA. What I could do is take out the tests from travis since it seems to be working fine on GA and leave (for the moment) the deployments on travis.

**Comment by lasofivec** [comment]
If you rather no have any overlap between travis and GA, I can make a few changes to the GA and make the PR only when travis can be completely suppressed
But in my experience there might be some oversighted issues with the deployment and I think is better at least for one release to have both travis and GA

**Comment by Didou09** [comment]
I agree with @lasofivec , even if it might be better on the long term to have a unique testing / deployment workflow, I also think we should go smoothly with this migration and have the overlap for at least one release to make sure we understand all possible issues and take the time to fix everything. Better be safe and keep the working solution as long as everything is not rock-solid and tested with GA, in real conditions (i.e.: with a real release).

File: .github/workflows/complete-matrix.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest coverage
+        pip install -r requirements.txt
+    - name: Lint with flake8
```
Comment (Didou09) [inline .github/workflows/complete-matrix.yml:29]: Good initiative 👍

File: .github/workflows/python-package-conda.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        conda install flake8
+        # stop the build if there are Python syntax errors or undefined names
+        # too many F82 errors, should uncomment the following line
+        # flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        flake8 . --count --select=E9,F63,F7 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: install tofu
+      run: |
+        pip install -e ".[dev]"
```
Comment (Didou09) [inline .github/workflows/python-package-conda.yml:41]: This file is called conda but the tofu installation is done with pip, I'm not sure I get the point ?

File: tofu/_plot.py [hunk 3]
```diff
@@ -87,8 +87,8 @@ def _plot_shotoverview_init(ns=1, sharet=True, sharey=True, shareRZ=True,
         else:
             laxt[ii] = fig.add_subplot(axarr[ii,:2], sharex=sht, sharey=shy)
             laxc[ii] = fig.add_subplot(axarr[ii,2], sharex=shRZ, sharey=shRZ)
-            if not shareRZ:
-                ax2.set_aspect('equal', adjustable='datalim')
+            # if not shareRZ:
```
Comment (Didou09) [inline tofu/_plot.py:90]: Reason for this commenting ? was it the source of a bug ?

**Review by Didou09 — COMMENTED** [review]
Good, but a few questions / comments below 👍

File: .github/workflows/python-package-conda.yml (hunk 2 above)
Comment (Didou09) [reply .github/workflows/python-package-conda.yml:41]: pip ? conda ? 
If you want to install from a conda package you first need to create the pack:
 conda build conda_recipe

And then install from that package
conda install -c conda-bld/<name of the created package>

**Review by Didou09 — COMMENTED** [review]
More comments / questions

File: tofu/_plot.py (hunk 3 above)
Comment (lasofivec) [reply tofu/_plot.py:90]: Yes, sorry I will uncomment it as I treated in the PR: #456

Comment (lasofivec) [reply tofu/_plot.py]: (`ax2` is undefined)

File: .github/workflows/python-package-conda.yml (hunk 2 above)
Comment (lasofivec) [reply .github/workflows/python-package-conda.yml:41]: mmm you are right, I could build the library with conda, however our `conda-recipe` includes the deployment (we have a `conda-upload.sh` file). I decided to build it with `pip` for simplicity. I am going to rename the file and not use `conda`, would you be ok with that solution ? I still have to write a GA file for conda deployment.

**Comment by lasofivec** [comment]
If you both agree, I think it's ready to be merged...?

File: .github/workflows/python-package-conda.yml (hunk 2 above)
Comment (Didou09) [reply .github/workflows/python-package-conda.yml:41]: OK for that solution

**Comment by Didou09** [comment]
Could you just do the renaming of the file (conda vs pip), for consistency ?
And Happy New Year !!! :-)

**Comment by lasofivec** [comment]
done, happy new year to you too :)

**Comment by lasofivec** [comment]
>     * Github actions deployment:
>       Trying pypi seems to be working for test pypi.
>       To be able to test it on tofu for sure, we need to add two  add tokens for PyPi on repo's secret ([see this doc](https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#saving-credentials-on-github))

Regarding the next release, I wrote here the link that I sent you earlier @Didou09 
From the messages I don't feel like we addressed this issue

**Pull request merged**